src/cargo/ops/cargo_clean.rs

```diff
@@ -250,31 +251,40 @@ fn clean_specs(
             }
         }
     } else {
-        // Try to reduce the amount of times we iterate over the same target directory by storing away
-        // the directories we've iterated over (and cleaned for a given package).
-        let mut cleaned_packages: HashMap<_, HashSet<_>> = HashMap::default();
+        let mut dirs_to_clean = DirectoriesToClean::default();
         for pkg in packages {
-            let pkg_dir = format!("{}-*", pkg.name());
+            let pkg_fingerprint_prefix = format!("{}-", pkg.name());
             clean_ctx.progress.on_cleaning_package(&pkg.name())?;
 
             // Clean fingerprints.
+            let package = &*pkg.name();
```

Comment:
How come we needed this?

Reply:
That's due to deref coercion not taking place for operands of `==`. I can inline the derefs if you prefer it that way.

Reply:
I find having `package` and `pkg` with different meanings confusing and have a slight preference for just inlining.